src/faostat_data_primap/download.py

```diff
@@ -0,0 +1,149 @@
+"""Downloads data from FAOSTAT website."""
+
+import pathlib
+import time
+import zipfile
+from datetime import datetime
+
+import bs4
+import requests
+from bs4 import BeautifulSoup
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+
+from faostat_data_primap.exceptions import DateTagNotFoundError
+
+
+def get_html_content(url: str) -> bs4.BeautifulSoup:
+    """
+    Get html from url.
+
+    Parameters
+    ----------
+    url
+        The url to the domain overview website.
+
+    Returns
+    -------
+        html content
+    """
+    # If the chrome driver isn't found on your system PATH, Selenium
+    # will automatically download it for you. Make sure there is no
+    # chromedriver installed on your system.
+    service = Service()
+    driver = webdriver.Chrome(service=service)
+
+    driver.get(url)
+
+    # give time to load javascript
+    time.sleep(3)
+
+    html_content = driver.page_source
+
+    return BeautifulSoup(html_content, "html.parser")
+
+
+def get_last_updated_date(soup: bs4.BeautifulSoup, url: str) -> str:
+    """
+    Get the date when data set way last updated from html text
+
+    The FAO stat domain overview page includes a date when
+    the data set was last updated. We need it to label our downloaded
+    data sets. This function searches and extracts the date
+    from the html code.
+
+    Parameters
+    ----------
+    soup
+        The beautiful soup object with all html code of the domain
+        overview page.
+    url
+        The url to the domain overview page.
+
+    Returns
+    -------
+        date when data set was last updated
+
+    Raises
+    ------
+    DateTagNotFoundError
+        If the tag for the date is not found in the html code
+    """
+    date_tag = soup.find("p", {"data-role": "date"})
+
+    if not date_tag:
+        raise DateTagNotFoundError(url=url)
+
+    last_updated = date_tag.get_text()
+    last_updated = datetime.strptime(last_updated, "%B %d, %Y").strftime("%Y-%m-%d")
+    return last_updated
+
+
+def download_file(url_download: str, save_path: pathlib.PosixPath):
+    """
+    Download file.
+
+    If an existing file is found at this location, the download is skipped.
+
+    Parameters
+    ----------
+    url_download
+        Remote URL to download the file from
+    save_path
+        Path to save the downloaded file to
+
+    Returns
+    -------
+        True if the file was downloaded, False if a cached file was found
```

Comment:
In case there is no file, but download fails, requests throws and error, correct?

Reply:
Pretty sure it does. I can try and make a test for that